autosubmit/platforms/platform.py

```diff
@@ -862,6 +867,25 @@ def connect(self, as_conf, reconnect=False):
     def restore_connection(self, as_conf):
         raise NotImplementedError
 
+    def clean_log_recovery_process(self) -> None:
+        """
+        Cleans the log recovery process variables.
+
+        This method sets the cleanup event to signal the log recovery process to finish,
+        waits for the process to join with a timeout, and then resets all related variables.
+        """
+        self.cleanup_event.set()  # Indicates to old child ( if reachable ) to finish.
+        if self.log_recovery_process:
+            self.log_recovery_process.join(timeout=60)  # Waits for old child ( if reachable ) to finish. Timeout in case of it being blocked.
+        # Resets everything related to the log recovery process.
+        self.recovery_queue = UniqueQueue()
+        self.log_retrieval_process_active = False
+        self.remove_workers(self.work_event)
+        self.work_event = Event()
+        self.cleanup_event = Event()
+        self.log_recovery_process = None
+        self.processed_wrapper_logs = set()
```

Comment:
:+1: I compared with the variables initialized in the constructor, and I think the only differences with the variables under `# Retrieval log process variables` are the `main_process_id` that's not reset, and the `max_waiting_jobs` & `keep_alive_timeout` as I think these shouldn't change/matter.